File: Advanced Business With .NET/sprint-1/src/Domain/Entities/FormularioDetalhado.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace UserApi.Domain.Entities
{
    public class FormularioDetalhado
    {
        [BsonId]
        public ObjectId Id { get; set; }
        public string? IdFormulario { get; set; }
        public string IdCliente { get; set; } = null!;
        public string IdEstadoCivil { get; set; } = null!;
        public string Historico_familiar { get; set; } = null!;
        public string Profissao { get; set; } = null!;
        public DateTime Data_ultima_atualizacao { get; set; }

        public string Data => ConvertToSaoPauloTime(Data_ultima_atualizacao).ToString("yyyy-MM-dd");

        public string Hora => ConvertToSaoPauloTime(Data_ultima_atualizacao).ToString("HH:mm:ss");

        private DateTime ConvertToSaoPauloTime(DateTime utcDateTime)
        {
            TimeZoneInfo saoPauloTimeZone = TimeZoneInfo.FindSystemTimeZoneById("E. South America Standard Time");
            return TimeZoneInfo.ConvertTimeFromUtc(utcDateTime, saoPauloTimeZone);
        }

        public float Renda_mensal { get; set; }
        public string Historico_medico { get; set; } = null!;
        public string Alergia { get; set; } = null!;
        public string Condicao_preexistente { get; set; } = null!;
        public string Uso_medicamento { get; set; } = null!;
        public string Familiar_com_doencas_dentarias { get; set; } = null!;
        public string Participacao_programa_preventivos { get; set; } = null!;
        public string Contato_emergencial { get; set; } = null!;
        public string Pesquisa_satisfacao { get; set; } = null!;
        public string Frequencia_consulta_periodica { get; set; } = null!;
        public string Sinalizacao_risco { get; set; } = null!;
        public string Historico_viagem { get; set; } = null!;
        public string Historico_mudanca_endereco { get; set; } = null!;
        public string Preferencia_contato { get; set; } = null!;

    }
}

File: JAVA ADVANCED/sprint-2/challenge/src/main/java/br/com/fiap/challenge/domains/FormularioDetalhado.java
package br.com.fiap.challenge.domains;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "formulario_detalhado")
public class FormularioDetalhado {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id_formulario", nullable = false)
    private String idFormulario;

    @ManyToOne
    @JoinColumn(name = "id_cliente", nullable = false)
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "id_estado_civil", nullable = false)
    private EstadoCivil estadoCivil;

    @Column(name = "historico_familiar", length = 250)
    private String historicoFamiliar;

    @Column(name = "profissao", length = 100)
    private String profissao;

    @Column(name = "renda_mensal")
    private Double rendaMensal;

    @Column(name = "historico_medico", length = 250)
    private String historicoMedico;

    @Column(name = "alergia", length = 250)
    private String alergia;

    @Column(name = "condicao_preexistente", length = 250)
    private String condicaoPreexistente;

    @Column(name = "uso_medicamento", length = 250)
    private String usoMedicamento;

    @Column(name = "familiar_com_doencas_dentarias", length = 255)
    private String familiarComDoencasDentarias;

    @Column(name = "participacao_em_programas_preventivos", length = 1)
    private Character participacaoEmProgramasPreventivos;

    @Column(name = "contato_emergencial", length = 250)
    private String contatoEmergencial;

    @Column(name = "pesquisa_satisfacao", length = 1)
    private Character pesquisaSatisfacao;

    @Column(name = "data_ultima_atualizacao")
    private LocalDate dataUltimaAtualizacao;

    @Column(name = "frequencia_consulta_periodica", length = 1)
    private Character frequenciaConsultaPeriodica;

    @Column(name = "sinalizacao_de_risco", length = 250)
    private String sinalizacaoDeRisco;

    @Column(name = "historico_de_viagem", length = 250)
    private String historicoDeViagem;

    @Column(name = "historico_de_mudancas_de_endereco", length = 250)
    private String historicoDeMudancasDeEndereco;

    @Column(name = "preferencia_de_contato", length = 50)
    private String preferenciaDeContato;

}
